src/views/catalog/CreateFromInstanceTypes/state/hooks/usePreferences.ts

```diff
@@ -0,0 +1,24 @@
+import { VirtualMachineClusterPreferenceModelGroupVersionKind } from '@kubevirt-ui/kubevirt-api/console';
+import { V1beta1VirtualMachineClusterPreference } from '@kubevirt-ui/kubevirt-api/kubevirt';
+import { isEmpty } from '@kubevirt-utils/utils/utils';
+import { useK8sWatchResource } from '@openshift-console/dynamic-plugin-sdk';
+
+type UsePreferences = (
+  namespaced?: boolean,
+) => [preferences: V1beta1VirtualMachineClusterPreference[], loaded: boolean, loadError: any];
+
+const usePreferences: UsePreferences = (namespaced) => {
+  const [preferences, loaded, loadError] = useK8sWatchResource<
+    V1beta1VirtualMachineClusterPreference[]
+  >({
+    groupVersionKind: VirtualMachineClusterPreferenceModelGroupVersionKind,
+    isList: true,
+    namespaced,
+  });
+
+  const errorState = !loaded || loadError || isEmpty(preferences);
+
+  return [errorState ? [] : preferences, loadError ? true : loaded, loadError];
```

Comment:
isnt this what u are looking for?

return [preferences || [], loaded, loadError] ?